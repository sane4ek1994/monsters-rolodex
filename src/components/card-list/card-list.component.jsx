import React from 'react'

import { Card } from '../card/card'

import './card-list.styles.css'

export const CardList = ({ monsters }) => (
  <div className='card-list'>
    {monsters.map((monster, index) => (
      <Card key={index} monster={monster} />
    ))}
  </div>
)

// export class CardList extends Component {
//   render() {
//     const { monsters } = this.props

//     return (
// <div className='card-list'>
//   {monsters.map((monster, index) => (
//     <Card key={index} monster={monster} />
//   ))}
// </div>
//     )
//   }
// }
