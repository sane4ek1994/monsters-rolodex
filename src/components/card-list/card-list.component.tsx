import type { Monster } from '../../app/App'

import { Card } from '../card/card'

import './card-list.styles.css'

type CardListProps = {
  monsters: Monster[]
}

export const CardList = ({ monsters }: CardListProps) => (
  <div className='card-list'>
    {monsters.map((monster, index) => (
      <Card key={index} monster={monster} />
    ))}
  </div>
)
