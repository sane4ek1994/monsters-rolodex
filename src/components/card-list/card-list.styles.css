.card-list {
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(190px, 1fr));
  grid-gap: 20px;
}
