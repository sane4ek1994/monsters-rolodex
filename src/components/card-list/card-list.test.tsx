import { shallow, mount, render } from 'enzyme'
import toJson from 'enzyme-to-json'
import { CardList } from './card-list.component'

it('expect to render Card-list component', () => {
  const mockRobots = [
    {
      id: '1',
      name: 'John Snow',
      username: 'JohnLeviev',
      email: '124@.gmail.com'
    }
  ]

  const cardListComponent = shallow(<CardList monsters={mockRobots} />)
  expect(toJson(cardListComponent)).toMatchSnapshot()
})
