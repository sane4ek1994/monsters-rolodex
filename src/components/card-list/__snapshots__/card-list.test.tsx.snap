// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div
  className="card-list"
>
  <Card
    key="0"
    monster={
      Object {
        "email": "124@.gmail.com",
        "id": "1",
        "name": "John Snow",
        "username": "JohnLeviev",
      }
    }
  />
</div>
`;

exports[`expect to render Card-list component 1`] = `
<div
  className="card-list"
>
  <Card
    key="0"
    monster={
      Object {
        "email": "124@.gmail.com",
        "id": "1",
        "name": "John Snow",
        "username": "JohnLeviev",
      }
    }
  />
</div>
`;
